### ISSUE TITLE
Distribute with pre-built opentracing/opentelemetry support

### ISSUE THREAD

**Issue by dekimsey** [haproxytech/haproxy-docker-alpine#5]
This is a feature request asking that these (awesome) containers come with pre-built opentracing (opentelemetry at some later point when it's compatible, see haproxy/haproxy#1640) libraries. I'm not sure how one should best implement adding support for the actual tracing libraries, but currently compiling them and then recompiling haproxy is challenging. Thus far I've been unable to get it to work.

In my particular case I was trying to get jaeger support added to my 2.4 image and I haven't been able to get it to successfully compile.

**Comment by dkorunic** [comment]
@dekimsey Request makes sense, although we definitely wouldn't enable OpenTracing by default in a default HAProxy Docker Alpine image, since that would mean that everyone defaulting on our image including all derived products (Ingress Controller) would get larger and more complex base image. I will add this to a task queue and possibly create an additional flavour image.

**Comment by dekimsey** [comment]
@dkorunic thank you! 

I can certainly understand the concern for bloat. Open tracing's design with client specific libraries probably makes it rather annoying to have embedded in a project like this. If or when haproxy transition to Opentelemetry I *think* that gets a whole lot less painful. As becomes the one and only library that needs to be compiled in and the rest is configuration. Maybe that's something that could be revisited at that point if reasonable.

Would this possible flavor be replicated for the debian version? I didn't want to file identical tickets. I ended up having to switch to debian because alpine didn't like our older docker installs (hosts have docker <20.0).

It makes me wonder if there is room for instead making a "full" image that compiles/activates more of haproxy's feature set. I haven't looked in depth at the available/relevant flags here. To be honest, I've only needed to manipulate the image for trying to add jwt auth handling via lua (couldn't get that to work anyhow).

There is a similar ticket at docker-library/haproxy#165, but I don't know how ownership works there. 

I'm still trying to compile my own 2.4+Jaeger. I'll share how I went about it if I can get it all to work successfully.
